app/controllers/Checkout.scala

```diff
@@ -1,13 +1,67 @@
 package controllers
 
-import actions.CommonActions.GoogleAuthenticatedStaffAction
+import actions.CommonActions._
+import model.{AddressData, PaymentData, PersonalData, SubscriptionData}
 import play.api.mvc._
+import services.CheckoutService
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 object Checkout extends Controller {
 
-  def renderCheckout = GoogleAuthenticatedStaffAction(Ok(views.html.checkout.payment()))
+  import play.api.data.Forms._
+  import play.api.data._
+
+  val subscriptionForm = Form(mapping(
+    "" -> mapping(
+      "first" -> text,
```

Comment:
Can we split them into three separate variables and then link them in the `subscrioptionForm`? I think it will be easier to read and understand later.